### PR TITLE
[INTERN-10] Cover both Fragments with Android tests.

### PR DESCRIPTION
**Pull request by Strishwork**

### CODE REVIEW

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt [hunk 1]
```diff
@@ -66,7 +66,7 @@ class CountryDetailsViewModelTest {
         assertThat(actualState.countryDetails.flag, `is`("https://test.com"))
         assertThat(actualState.countryDetails.currencyNames[0], `is`("Test currency"))
         assertThat(actualState.countryDetails.languages[0], `is`("Test language"))
-        assertThat(actualState.countryDetails.timezones[0], `is`("UTC+03:00"))
+        assertThat(actualState.countryDetails.timezones[0], `is`("UTC+05:00"))
```
Comment (yeoliinyk) [inline app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Let's revert this.

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt [hunk 2]
```diff
@@ -133,7 +133,7 @@ class CountryDetailsViewModelTest {
             `when`(flag?.svgFile).thenReturn("https://test.com")
             `when`(currencies).thenReturn(listOf(CountryDetails.Currency("", "Test currency", "")))
             `when`(officialLanguages).thenReturn(listOf(CountryDetails.OfficialLanguage("", "Test language")))
-            `when`(timezones).thenReturn(listOf(CountryDetails.Timezone("", "UTC+03:00")))
+            `when`(timezones).thenReturn(listOf(CountryDetails.Timezone("", "UTC+05:00")))
```
Comment (yeoliinyk) [inline app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Same here

File: app/src/main/res/layout/country_preview_layout.xml [hunk 3]
```diff
@@ -2,7 +2,8 @@
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:id="@+id/previewRecyclerViewLayout">
```
Comment (yeoliinyk) [inline app/src/main/res/layout/country_preview_layout.xml]: Let's consider better naming here.

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/main"
```
Comment (yeoliinyk) [inline app/src/main/res/layout/activity_main.xml]: Let's rename.

File: app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt [hunk 5]
```diff
@@ -28,7 +28,9 @@ class CountryPreviewFragment : Fragment() {
 
     override fun onAttach(context: Context) {
         super.onAttach(context)
-        listener = context as MainActivity
+        if (context is MainActivity) {
+            listener = context
+        }
```
Comment (yeoliinyk) [inline app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt]: Cover this with tests also. Please investigate possible solutions.

File: app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt [hunk 6]
```diff
@@ -77,7 +77,9 @@ class CountryDetailsFragment : Fragment(), CountryDetailsAdapter.ViewHolder.OnIt
     }
 
     private fun showDialog(section: DetailsSections, dialogMessage: List<String>) {
```
Comment (yeoliinyk) [inline app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt]: [nit] We can refactor this as well and pass two strings for title and message.

File: app/build.gradle [hunk 7]
```diff
@@ -31,16 +35,19 @@ android {
...
 
+    def lifecycle_version = "2.2.0"
+    def dagger_version = "2.27"
+    def mockito_version = "3.4.6"
+
     implementation 'androidx.cardview:cardview:1.0.0'
     implementation 'androidx.recyclerview:recyclerview:1.1.0'
-    def lifecycle_version = "2.2.0"
-    def dagger_version = "2.25.2"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (yeoliinyk) [inline app/build.gradle]: Let's check if we really need this?

**Review by yeoliinyk — COMMENTED** [review]
Looks good but there is at least one missed test.

File: app/build.gradle (hunk 7 above)
Comment (Strishwork) [reply app/build.gradle]: Removed.

File: app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt (hunk 6 above)
Comment (Strishwork) [reply app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt]: Done.

File: app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt (hunk 5 above)
Comment (Strishwork) [reply app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt]: Done.

File: app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (Strishwork) [reply app/src/main/res/layout/activity_main.xml]: Renamed.

File: app/src/main/res/layout/country_preview_layout.xml (hunk 3 above)
Comment (Strishwork) [reply app/src/main/res/layout/country_preview_layout.xml]: Renamed.

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt (hunk 2 above)
Comment (Strishwork) [reply app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Removed.

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt (hunk 1 above)
Comment (Strishwork) [reply app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Removed.

**Review by yeoliinyk — APPROVED** [review]

**Pull request merged**